modules/apps/portal/portal-servlet-test/src/testIntegration/java/com/liferay/portal/servlet/filters/multihost/locale/test/MultiHostLocaleFilterTest.java

```diff
@@ -0,0 +1,238 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.servlet.filters.multihost.locale.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.exception.AvailableLocaleException;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.model.LayoutSet;
+import com.liferay.portal.kernel.model.VirtualHost;
+import com.liferay.portal.kernel.service.LayoutSetLocalService;
+import com.liferay.portal.kernel.service.VirtualHostLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.servlet.filters.absoluteredirects.AbsoluteRedirectsFilter;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.TreeMap;
+
+/**
+ * @author Noemi Zamarripa
+ */
+@RunWith(Arquillian.class)
+public class MultiHostLocaleFilterTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws PortalException {
+		_groupId = TestPropsValues.getGroupId();
+		_availableLocales = LanguageUtil.getAvailableLocales(_groupId);
+		_layoutSetId = _layoutSetLocalService.getLayoutSet(
+			_groupId, false).getLayoutSetId();
+	}
+
+	@After
+	public void tearDown() throws PortalException {
+		for (String hostname : _treeMap.keySet()) {
+			VirtualHost host = _virtualHostLocalService.getVirtualHost(hostname);
+
+			_virtualHostLocalService.deleteVirtualHost(host);
+		}
+	}
+
+	@Test
+	public void testMultiHostNameAndLocale1() throws Exception {
+
+		_treeMap.put(_TESTHOST_FR, _FR_LANGUAGE_ID);
+		_treeMap.put(_TESTHOST_JP, _JP_LANGUAGE_ID);
+
+		_layoutSetLocalService.updateVirtualHosts(
+			_groupId, false, _treeMap);
+
+		for (String host : _treeMap.keySet()) {
+			_setupRequest(host, _treeMap.get(host));
+
+			_invokeRequestFilter();
+
+			Assert.assertEquals(
+				_mockHttpServletRequest.getAttribute(WebKeys.I18N_LANGUAGE_ID),
+				_virtualHostLocalService.getVirtualHost(host).getLanguageId());
```

Comment:
Just some points on form.

In order for JUnit reports to make sense, the expected results must be on the left.